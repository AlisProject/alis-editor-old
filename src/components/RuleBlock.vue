<template>
  <div class="rule">
    <hr/>
    <textarea @keydown="handleKeydown" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  methods: {
    handleKeydown(event: KeyboardEvent) {
      if (event.keyCode === 8) {
        this.$emit('delete')
      }
    }
  }
})
</script>

<style scoped>
.rule {
  position: relative;
  padding: 10px 0;
}

.rule hr {
  margin: 0;
}

.rule textarea {
  color: transparent;
  background: transparent;
  resize: none;
  opacity: 0;
  outline: none;
  display: block;
  position: absolute;
  z-index: 100000;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}
</style>
