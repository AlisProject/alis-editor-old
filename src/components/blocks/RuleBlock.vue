<template>
  <div class="rule" v-if="!preview">
    <hr />
    <ShadowInput @delete="handleDelete" @addimageuri="handleAddImage" />
  </div>
  <div class="rule" v-else><hr /></div>
</template>

<script lang="ts">
import Vue from 'vue'
import ShadowInput from '../blocks/ShadowInput.vue'

export default Vue.extend({
  components: {
    ShadowInput
  },
  props: {
    block: Object,
    preview: {
      default: false,
      type: Boolean
    }
  },
  methods: {
    handleDelete() {
      this.$emit('delete', this.block)
    },
    handleAddImage(src: string) {
      this.$emit('addimageuri', src)
    }
  }
})
</script>

<style scoped>
.rule {
  position: relative;
  padding: 10px 0;
}

.rule hr {
  margin: 0;
  border: none;
  border-top: dotted 3px #e3e3e3;
  height: 2px;
}

.rule textarea {
  color: transparent;
  background: transparent;
  resize: none;
  opacity: 0;
  outline: none;
  display: block;
  position: absolute;
  z-index: 100000;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

@media (max-width: 768px) {
  .rule {
    margin: 0 8px;
  }
}
</style>
