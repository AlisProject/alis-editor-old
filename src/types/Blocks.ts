export enum BlockType {
  Rule = 'Rule',
  Text = 'Text',
  Paragraph = 'Paragraph',
  Image = 'Image',
  Quote = 'Quote',
  Heading = 'Heading',
  Embed = 'Embed'
}

export interface Block {
  id: string
  type: BlockType
  payload?: any
  children?: Block[]
}

export interface RuleBlock extends Block {
  type: BlockType.Rule
}

export interface EmbedBlock extends Block {
  type: BlockType.Embed,
  payload: {
    src: string
  }
}

export interface ParagraphBlock extends Block {
  type: BlockType.Paragraph
  payload: {
    body: string
  }
}

export interface TextBlock extends Block {
  type: BlockType.Text
  payload: {
    body: string
  }
}

export interface ImageBlock extends Block {
  type: BlockType.Image
  payload: {
    src: string
    align?: string
  }
  children: Block[]
}

export interface HeadingBlock extends Block {
  type: BlockType.Heading
  payload: {
    size: 'h2' | 'h3'
    body: string
  }
  children: Block[]
}

export interface QuoteBlock extends Block {
  type: BlockType.Quote
  children: Block[]
}

/*
「詳細は [こちら](https://example.com) から確認できます」を再現する場合
[
  {
    type: 'Paragraph',
    children: [
      {
        type: 'Text',
        payload: {
          body: '詳細は'
        }
      },
      {
        type: 'Link',
        payload: {
          href: 'https://example.com'
        },
        children: [
          {
            type: 'Text',
            payload: {
              body: 'こちら'
            }
          }
        ]
      },
      {
        type: 'Text',
        payload: {
          body: 'から確認できます'
        }
      }
    ]
  }
]

*/
