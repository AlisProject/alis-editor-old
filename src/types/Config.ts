export interface AlisEditorConfig {
  articleId: string
  preview: boolean
  iframelyApikey: string
  uploadEndpoint: string
}
