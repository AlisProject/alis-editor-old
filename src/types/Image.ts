export interface ImageUploadRequest {
  article_image: string
}

export interface ImageUploadResponse {
  image_url: string
}
