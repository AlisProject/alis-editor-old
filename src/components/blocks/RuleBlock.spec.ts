import { mount,config } from '@vue/test-utils'
import RuleBlock from './RuleBlock.vue'
import * as BlockTypes from '../../types/Blocks'
config.logModifiedComponents = false

describe('RuleBlock.vue', () => {
  test('renders correctly', () => {
    const block: BlockTypes.RuleBlock = {
      id: '123',
      type: BlockTypes.BlockType.Rule
    }
    const wrapper = mount(RuleBlock, {
      propsData: {
        block
      },
      stubs: ['ShadowInput']
    })
    expect(wrapper).toMatchSnapshot()
  })

  test('renders correctly(preview mode)', () => {
    const block: BlockTypes.RuleBlock = {
      id: '123',
      type: BlockTypes.BlockType.Rule
    }
    const wrapper = mount(RuleBlock, {
      propsData: {
        block,
        preview: true
      },
      stubs: ['ShadowInput']
    })
    expect(wrapper).toMatchSnapshot()
  })
})
