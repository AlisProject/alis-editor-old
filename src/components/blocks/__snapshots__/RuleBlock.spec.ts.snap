// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleBlock.vue renders correctly 1`] = `
<div class="rule">
  <hr>
  <shadowinput-stub></shadowinput-stub>
</div>
`;

exports[`RuleBlock.vue renders correctly(preview mode) 1`] = `
<div class="rule">
  <hr>
</div>
`;
